source/sessions/driver-sessions.rst

```diff
@@ -444,7 +444,9 @@ Check Whether a Deployment Supports Session). If sessions are supported, the
 driver MUST behave as if a new ``ClientSession`` was started just for this one
 operation and ended immediately after this operation completes. The actual
 implementation will likely involve calling ``client.startSession``, but that is not
-required by this spec.
+required by this spec. Regardless, please consult the startSession section to
+replicate the required steps for creating a session.
+Drivers MUST create an implicit session only after successfully checking out a connection.
```

Comment:
The C# driver will have to continue to create an implicit session *before* checking out a connection, but we think we can achieve the desired result of not consuming a server session id until after the connection is checked out by making the implicit session lazily acquire a server session id at a later point in time.

Not sure if you want the spec to be as restrictive as currently written here because some drivers might actually not be able to do it exactly this way.

Reply:
Maybe the wording could be "MUST NOT consume a server session id until after the connection is checked out" instead of "MUST NOT create an implicit session until after the connection is checked out"?

Reply:
Yea I think this is a good idea, and I can mention the reasoning in Q&A since other drivers may approach this with a similar design. Updated!